### DISCUSSION TITLE
Cycle through svg color palette

### DISCUSSION THREAD

**Discussion by c64zottel** [Ogeon/palette#329] [Q&A] [answered]
Is there a way to cycle through the named colors?

I would argue a crate called palette should provide such functionality. Probably with different palettes preset (SVG, terminal..).

My use case: I like to upload such palettes to the gpu.

Did I miss something in the docs?

**Comment by Ogeon** [comment]
Hi! There's currently nothing implemented for it but it wouldn't be difficult to have a constant slice with all of them. Either with or without the names. Or both.

As for having other palettes, I would like to avoid having a curated palette collection (despite the crate's name 😅). That's better as an external resource or different crate. The SVG colors were added long ago because they are fairly ubiquitous and standardized, so they work a bit like a `from_str` option. If there's another standard like it, then maybe that could be added, but I'm not sure I would have added even the SVG colors today.

**Reply by c64zottel** [reply]
Hey, thanks for your message.
While I honor your opinion, may I ask if there is a deeper reasoning?

I really think that palette would broaden it's user base. That is because there is (to the best to my knowledge) no crate combining functionality provided by palette as well as a decent palette of predefined colors. (There is no special reason me using palette palette-wise - not sorry about it ;))

An extension like this could be hidden behind a feature, so no bloating.

Another standard would be: https://en.wikipedia.org/wiki/X11_color_names
I would offer a patch in case you change your mind.

**Reply by Ogeon** [reply, marked as answer]
It's really just a matter of staying focused on the color math. 🙂 It's not so much about what gets compiled into the program, even though it's more for the compiler to process, but more about maintenance and purpose.

Regarding maintenance, it would be more to test, organize and take into account when doing changes. In this case it would mostly be a matter of curating the collection, since an array of colors usually doesn't cause much issues once it's added. Unless something like `phf`, that's used for the string lookup, suddenly breaks compatibility. 😅 

The reason is more about purpose. This crate's main purpose is color math and I don't want to widen that scope too much. If something ends up being a topic on its own, like the gradients that used to be in there up until 0.7.0, I tend to cut it out as soon as a better option exists. 🚮 So while it's convenient to have a single source of everything, I think it's a bit too wide for this one crate. 🤔

The SVG colors were added back when the crate hadn't found its current form. It was a more opinionated, so a set of color presets made sense back then. Here's the documentation for version 0.2.1, where it was added: https://docs.rs/palette/0.2.1. Note the `Color` enum, which is gone now, that used to change its inner state for different operations. A lot of other things worked differently too. See if you can find sRGB, for example. 😄 

I wouldn't be entirely opposed to something like a `palette-palettes` crate, for example, that would only meant to collect nice palettes. That would be really neat, in fact 👀, except I don't think I would have the personal bandwidth right now to go out and find or create them myself. 😅
